gems/pending/metadata/linux/OpenstackConfigurationParser.rb

```diff
@@ -0,0 +1,200 @@
+          # non commented value has always precedence over commented. Name and Section are an unique identifier inside
+          # of the file.
+          attribute[:name]               = match[1]
+          attribute[:value]              = match[2]
+          attribute[:value_interpolated] = match[2].dup
+          attribute[:section]            = section
+          # Allowed values of source are ['defined', 'default'], defined says the value was actually defined in conf
+          # file, default says it is present only as a comment, that shows what should be the default value, if it's not
+          # redefined. It is based on commented_line boolean
+          attribute[:source]             = nil
+          commented_line                 = line.match(COMMENTED_LINE_REGEXP)
+
+          if (existing_attribute = attributes_hash.fetch_path(section, attribute[:name]))
+            # If it's commented attribute, we will not change value of existing attribute
+            unless commented_line
+              existing_attribute[:value]              = attribute[:value]
+              existing_attribute[:value_interpolated] = attribute[:value_interpolated]
+              existing_attribute[:source] = 'defined'
+            end
+          else
+            # New attribute, just add it whole
+            attribute[:source] = commented_line ? 'default' : 'defined'
+            (attributes_hash[section] ||= {})[attribute[:name]] = attribute
+          end
+
+          unless commented_line
+            # Placeholder for continuation line, only not commented lines can have continuation lines
+            last_attribute_line_indent = count_leading_spaces(line)
+            last_attribute = attribute
+          end
+          # Start over after creating each attribute
+          attribute      = {}
+        elsif (match = line.match(DESCRIPTION_REGEXP))
+          # If line starts with comment and it's not attribute, it's description of the attribute
+          next if line.match(HEADER_REGEXP) # don't put headers into description
+          attribute[:description] ||= ""
+          attribute[:description] += "\n" unless attribute[:description].blank?
+          attribute[:description] += match[1]
+        else
+          # New line or unknown, reset attribute and start over
+          last_attribute = {}
+          attribute      = {}
+        end
+      end
+
+      make_interpolation!(attributes_hash)
+
+      # Convert nested hashes to flat list of attributes
+      attributes_hash.values.each_with_object([]) { |x, obj| obj.concat(x.values) }
+    end
+
+    def make_interpolation!(attributes_hash)
+      # As specified in python doc, there is special section named 'DEFAULT', that has default values for all sections
+      default_section_hash = attributes_hash['DEFAULT'] || {}
+
+      # Functionality of python's class configparser.BasicInterpolation
```

Comment:
this one seems like a rogue comment, but I am doing 
configparser.BasicInterpolation, configparser.ExtendedInterpolation and some non documented type, as stated below

given it's few lines, seems to me it's better like this, rather than combining custom parser with calling of some python classes.